### PR TITLE
Deprecate fake UPS

### PR DESCRIPTION
**Pull request by mugling**
Factored out from #16829

### CODE REVIEW

**Comment by Coolthulhu** [comment]
This doesn't fix the "attempted to fire zero or negative charges from laser rifle". I can't really test it until that bug goes away.

**Comment by mugling** [comment]
That involves refactoring of the turret code which is easier to do with this PR out of the way. You could temporarily falsify the charges at the point of that debugmsg to ensure no regression. I appreciate it's not ideal but I don't want to go through the entire turret code in one pass as opposed to smaller chunks.

**Comment by Coolthulhu** [comment]
What I can tell from the testing is that with this PR, the guns that would use UPS don't work at all, while before it, they would debugmsg and then shoot anyway.

Still, I can't really say this PR is working when it depends on a major bug being fixed to really be testable. Is there no way to hack around this here? I'll look at the code and try to suggest at least a temporary hack.

**Comment by mugling** [comment]
Possible solution is in manual_fire_turret

**Comment by mugling** [comment]
Try af7e4a4. The proper fix is to have `cycle_turret_mode` use proper gun modes.

**Comment by Coolthulhu** [comment]
How about this:
* When installing a turret, its item gains a "TURRET" flag
* When calling `fire_gun` on an item that has the "TURRET" flag, it drains charges from the current vehicle to power self
* When removing the turret item, the "TURRET" flag is removed

**Comment by mugling** [comment]
I've gone with the more generic `VEHICLE` flag for an item set on installation and unset on removal. The issue with negative charges is separate and to do with problems with setting the firing mode.

**Comment by Coolthulhu** [comment]
Still nothing

**Comment by mugling** [comment]
#17017 includes a fix to set `VEHICLE` flag on game load which isn't pushed to this branch. Try removing and reinstalling the turret

**Comment by Coolthulhu** [comment]
For some reason the section with "VEHICLE" flag is ignored - `shots` is 5 before it, 0 after.

**Comment by mugling** [comment]
Know cause. The gun is not a duplicate of `base` and therefore is missing the flag

**Comment by mugling** [comment]
See 1346bb0

File: src/mattack_actors.cpp [hunk 1]
```diff
@@ -411,7 +411,7 @@ void gun_actor::shoot( monster &z, Creature &target, const std::string &mode ) c
     }
 
     tmp.weapon = gun;
-    tmp.worn.push_back( item( "fake_UPS", calendar::turn, 1000 ) );
+    tmp.i_add( item( "UPS", calendar::turn, 1000 ) );
```
Comment (Coolthulhu) [inline src/mattack_actors.cpp]: Crap, I merged too soon before I tested this. "UPS" is not an item, "UPS_off" or something like that is the item we want.

Can you fix it in #17020? That PR depends on UPS spawning right to test solar turrets.

Comment (mugling) [inline src/mattack_actors.cpp]: Yes, I'll push that now

**Pull request merged**